### PR TITLE
Refactoring logs in the auth rest api. Error logs changed to debug logs

### PR DESCRIPTION
**Pull request by Niveathika92**
## Purpose
Refactoring logs in the auth rest api. Error logs changed to debug logs

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: components/org.wso2.carbon.analytics.auth.rest.api/src/main/java/org/wso2/carbon/analytics/auth/rest/api/impl/LoginApiServiceImpl.java [hunk 1]
```diff
@@ -81,8 +81,10 @@ public Response loginAppNamePost(String appName
                 refToken = AuthUtil
                         .extractTokenFromHeaders(request.getHeaders(), AuthRESTAPIConstants.WSO2_SP_REFRESH_TOKEN);
                 if (refToken == null) {
-                    LOG.error("Unable to extract refresh token from the header for the request '"
-                            + removeCRLFCharacters(appName));
+                    if (LOG.isDebugEnabled()) {
```
Comment (this) [inline components/org.wso2.carbon.analytics.auth.rest.api/src/main/java/org/wso2/carbon/analytics/auth/rest/api/impl/LoginApiServiceImpl.java]: No need for check `isDebugEnabled()` with SLF4J

Comment (this) [reply components/org.wso2.carbon.analytics.auth.rest.api/src/main/java/org/wso2/carbon/analytics/auth/rest/api/impl/LoginApiServiceImpl.java]: Please use "parameterized" log messages. Refer example [here](https://github.com/this/carbon-ui-server/blob/ebae0947615eabb91bfa5fdf3aa8cc6e78fc21ad/components/org.wso2.carbon.uiserver/src/main/java/org/wso2/carbon/uiserver/internal/StartupListener.java#L89)

Comment (suhothayan) [reply components/org.wso2.carbon.analytics.auth.rest.api/src/main/java/org/wso2/carbon/analytics/auth/rest/api/impl/LoginApiServiceImpl.java]: I think we have to else we’ll do unnecessary string concatunations

Comment (this) [reply components/org.wso2.carbon.analytics.auth.rest.api/src/main/java/org/wso2/carbon/analytics/auth/rest/api/impl/LoginApiServiceImpl.java]: If you use parameterized messages, then any string concatenations won't happen.
`logger.debug("Student '{}' is excellent.", student.getName());`
more info: https://www.slf4j.org/faq.html#logging_performance

**Comment by suhothayan** [comment]
Why are we removing isdebugeenabled ?shouldnt that reduces the cost of string manipulations ?

File: components/org.wso2.carbon.analytics.auth.rest.api/src/main/java/org/wso2/carbon/analytics/auth/rest/api/impl/LoginApiServiceImpl.java (hunk 1 above)
Comment (Niveathika92) [reply components/org.wso2.carbon.analytics.auth.rest.api/src/main/java/org/wso2/carbon/analytics/auth/rest/api/impl/LoginApiServiceImpl.java]: Fixed in 96ca4ef and 2882642

**Comment by this** [comment]
@suhothayan, We use SLF4J as the logging framework.
- SLF4J internally checks whether debug level is enabled or not, hence we don't need to check.
- SLF4J has "parameterized" log messages. With parameterized messages, any string concatenation won't happen if debug level is not enabled.

Instead of
```java
if(logger.isDebugEnabled()) {
     logger.debug("Student '" + student.getName() + "' is excellent.");
}
```
with SLF4J we can do 
```java
logger.debug("Student '{}' is excellent.", student.getName());
```
Note the **{}** placeholder.

Read more: https://www.slf4j.org/faq.html#logging_performance

**Comment by this** [comment]
@Niveathika92, I agree with @suhothayan's last remark. Shall we wrap logs with `if(logger.isDebugEnabled()) {` where u have called the `removeCRLFCharacters` method for log messages

**Comment by suhothayan** [comment]
@sajithAR thanks for the clarification but as I have commented since CRLF function it s of complex and As it’s been callled for each log  we’ll go with is debug check for now.

**Comment by Niveathika92** [comment]
@suhothayan  @sajithar(@this) added the debugEnabled check in c8dad2f

**Pull request merged**